Sources/TSCBasic/Process.swift

```diff
@@ -554,19 +608,22 @@ public final class Process: ObjectIdentifierProtocol {
             stderrOutput: stderr.result
         )
         return executionResult
-      #else
-        return try serialQueue.sync {
-            precondition(launched, "The process is not yet launched.")
-
-            // If the process has already finsihed, return it.
-            if let existingResult = _result {
-                return existingResult
-            }
-
+        #else
+        self.stateLock.lock()
+        switch self.state {
+        case .idle:
+            defer { self.stateLock.unlock() }
+            preconditionFailure("The process is not yet launched.")
+        case .complete(let result):
+            return result
```

Comment:
This case seems to be missing a `self.stateLock.unlock()`.  Or is it intentional to return while holding the lock here?  If so, then that probably warrants a comment that it's on purpose.  But it doesn't look right since this is a public function and that would cause a subsequent access of the public `result` property to deadlock.

Reply:
good catch